### PR TITLE
Artifact SLF4J LOG4J-12 has been relocated

### LINKED ISSUE #20111: Artifact SLF4J LOG4J-12 has been relocated
### Description

When building Keycloak, this warning is shown to the user:
```
[WARNING] The artifact org.slf4j:slf4j-log4j12:jar:2.0.6 has been relocated to org.slf4j:slf4j-reload4j:jar:2.0.6
```

You can see it here: https://mvnrepository.com/artifact/org.slf4j/slf4j-log4j12

### PR DESCRIPTION
**Pull request by mabartos**
Closes #20111

@abstractj Could you please check it? Thanks

### CODE REVIEW

**Review by vmuzikar — DISMISSED** [review]

**Review by abstractj — COMMENTED** [review]
@mabartos changes overall look good, but I'm afraid that we need to keep those versions in sync with Quarkus, unless @vmuzikar says otherwise.

As a sanity check, I'd also recommend waiting for the feedback from the Store team.

File: pom.xml [hunk 1]
```diff
@@ -112,8 +112,8 @@
         <resteasy.version>6.2.3.Final</resteasy.version>
         <resteasy.undertow.version>${resteasy.version}</resteasy.undertow.version>
         <owasp.html.sanitizer.version>20211018.2</owasp.html.sanitizer.version>
-        <slf4j-api.version>2.0.6</slf4j-api.version>
-        <slf4j.version>2.0.6</slf4j.version>
+        <slf4j-api.version>2.0.7</slf4j-api.version>
```
[the remark this answers is not in the document]
Comment (mabartos) [reply pom.xml:115]: @abstractj Yep, you're right.

**Comment by mabartos** [comment]
@mhajas @ahus1 Are you ok with these changes?

**Review by abstractj — APPROVED** [review]
LGTM

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**